### ISSUE TITLE
Sylius Shop user resource declaration does not have an interface config

### ISSUE THREAD

**Issue by bitbager** [Sylius/Sylius#9360]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | yes
| BC Break report? | no
| RFC?             | no
| Sylius version   | 1.1.5

When I try to configure Doctrine mapping reference by interface for ShopUser, I am getting the class not found error:

```
$ bin/console do:sche:cre

 !                                                                                                                      
 ! [CAUTION] This operation should not be executed in a production environment!                                         
 !                                                                                                                      

 Creating database schema...


In MappingException.php line 96:
                                                                        
  Class 'Sylius\Component\Core\Model\ShopUserInterface' does not exis
```

This might be caused by missing interface config in `src/Sylius/Bundle/CoreBundle/Resources/config/app/sylius/sylius_user.yml` file.

Below is my Doctrine config. I hope this will give you a better understanding of the problem.

```
<?xml version="1.0" encoding="UTF-8"?>

<doctrine-mapping xmlns="http://doctrine-project.org/schemas/orm/doctrine-mapping"
                  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                  xsi:schemaLocation="http://doctrine-project.org/schemas/orm/doctrine-mapping
                                      http://doctrine-project.org/schemas/orm/doctrine-mapping.xsd">

    <mapped-superclass name="BitBag\SyliusWishlistPlugin\Entity\Wishlist" table="bitbag_wishlist">
        <id name="id" column="id" type="integer">
            <generator strategy="AUTO" />
        </id>

        <many-to-many field="products" target-entity="Sylius\Component\Product\Model\ProductInterface">
            <join-table name="bitbag_wishlist_products">
                <join-columns>
                    <join-column name="wishlist_id" referenced-column-name="id" nullable="false" on-delete="CASCADE" />
                </join-columns>
                <inverse-join-columns>
                    <join-column name="product_id" referenced-column-name="id" nullable="false" on-delete="CASCADE" />
                </inverse-join-columns>
            </join-table>
        </many-to-many>

        <one-to-one field="shopUser" target-entity="Sylius\Component\Core\Model\ShopUserInterface">
            <join-column name="shop_user_id" referenced-column-name="id" nullable="true" />
        </one-to-one>

    </mapped-superclass>

</doctrine-mapping>
```

**Comment by teohhanhui** [comment]
I think there is a design problem.

If you look at https://github.com/Sylius/SyliusResourceBundle/blob/v1.5.0/src/Bundle/DependencyInjection/Compiler/DoctrineTargetEntitiesResolverPass.php

it does not go up the inheritance chain - and it's not safe to do so, as we can't tell which interfaces should be mapped, and which should not. So if we set `interface` to a child interface, things would stop working if you try using the parent interface(s).

What is actually expected is that all of the interfaces should work, e.g.:

- `Sylius\Component\Product\Model\ProductInterface`
- `Sylius\Component\Core\Model\ProductInterface`
- `App\Model\ProductInterface` (for example)

So we need a way to map multiple interfaces.

**Comment by vvasiloi** [comment]
Defining the interfaces should solve the issue.
But as @bitbager pointed out, it should be done in https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/CoreBundle/Resources/config/app/sylius/sylius_user.yml

`config/packages/sylius_user.yaml`
```
sylius_user:
    resources:
        admin:
            user:
                classes:
                    interface: Sylius\Component\Core\Model\AdminUserInterface
        shop:
            user:
                classes:
                    interface: Sylius\Component\Core\Model\ShopUserInterface
        oauth:
            user:
                classes:
                    interface: Sylius\Component\User\Model\UserOAuthInterface
```

**Comment by kortwotze** [comment]
I ran into the same problem using the bitbag wishlist-plugin. Adding the proper interface classes to user entities solved the problem.

I think the solution of @vvasiloi should be implemented.

**Comment by kortwotze** [comment]
Guys, check the PR @pamil created for this :heart:

**Comment by pamil** [comment]
Fixed by #10450.
